recipes/carbon.rb

```diff
@@ -17,25 +17,29 @@
 # limitations under the License.
 #
 
-python_runtime 'carbons_python' do
-  provider :system
-  version '2.7'
-  options pip_version: true
+include_recipe 'graphite::_user'
+include_recipe 'graphite::_directories'
+
+pyenv_user_install 'carbons_pyenv' do
+  user node['graphite']['user']
 end
 
-include_recipe 'graphite::_user'
+pyenv_python node['graphite']['pyenv']['python_version'] do
+  user node['graphite']['user']
+end
 
-directory node['graphite']['base_dir'] do
-  owner node['graphite']['user']
-  group node['graphite']['group']
+pyenv_global node['graphite']['pyenv']['python_version'] do
+  user node['graphite']['user']
 end
 
-python_virtualenv node['graphite']['base_dir'] do
+pyenv_pip 'virtualenv' do
+  user node['graphite']['user']
+end
+
+pyenv_script 'setup graphite virtualenv' do
```

Comment:
Because this cookbook was using virtualenv via the `poise_python` cookbook for package installs/running I decided to stick with that for the switch to pyenv here. Its a little funky the way I'm setting the virtualenv via the `pyenv_script` resource but I had a lot of trouble getting pyenv's virtualenv plugin working. this seemed to be the easiest way to enable it.